src/Microsoft.ApiDesignGuidelines.Analyzers/Core/OverrideEqualsAndOperatorEqualsOnValueTypes.cs

```diff
@@ -1,49 +1,74 @@
-                                                                             customTags: WellKnownDiagnosticTags.Telemetry);
+                                                                             s_category,
+                                                                             s_severity,
+                                                                             s_isEnabledByDefault,
+                                                                             s_localizableDescription,
+                                                                             s_helpLinkUri,
+                                                                             s_customTags);
+
         internal static DiagnosticDescriptor OpEqualityRule = new DiagnosticDescriptor(RuleId,
                                                                              s_localizableTitle,
                                                                              s_localizableMessageOpEquality,
-                                                                             DiagnosticCategory.Performance,
-                                                                             DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
-                                                                             description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
-                                                                             customTags: WellKnownDiagnosticTags.Telemetry);
+                                                                             s_category,
+                                                                             s_severity,
+                                                                             s_isEnabledByDefault,
+                                                                             s_localizableDescription,
+                                                                             s_helpLinkUri,
+                                                                             s_customTags);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(EqualsRule, OpEqualityRule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
-            
+            analysisContext.RegisterSymbolAction(context =>
+            {
+                var namedType = (INamedTypeSymbol)context.Symbol;
+                if (namedType.IsValueType)
+                {
+                    if (!namedType.OverridesEquals())
+                    {
+                        context.ReportDiagnostic(
+                            Diagnostic.Create(EqualsRule, namedType.FirstLocation(), namedType.Name));
```

Comment:
You can instead write namedType.CreateDiagnostic(EqualsRule, namedType.Name) - this creates a diagnostic in the first location that's not in metadata as opposed to just the first location. (CreateDiagnostic is an extension method in Analyzer.Utilities)